### ISSUE TITLE
pipenv downgrades dependency after finding conflict resolution

### ISSUE THREAD

**Issue by ik-nitk** [pypa/pipenv#3025]
#### issue:
For example:
splinter depends on latest selenium > 3.5.  and pipenv successfully installs it.   But if we install a package like huxley which also depends on selenium < 2.5 , the pipenv install command does give error and halts, but it downgrades the version of selenium which was already installed.  

### Expected result
I was expecting the commands fails and the system would not have updated my already installed package. 

### Actual result

selenium got downgraded to lower version 

### Steps to replicate

````
$ pipenv graph 
splinter==0.9.0
  - selenium [required: >=3.14.0, installed: 3.14.1]
    - urllib3 [required: Any, installed: 1.23]

$ pipenv install huxley
Warning: Your dependencies could not be resolved. You likely have a mismatch in your sub-dependencies.
  First try clearing your dependency cache with $ pipenv lock --clear, then try the original command again.
 Alternatively, you can use $ pipenv install --skip-lock to bypass this mechanism, then run $ pipenv graph to inspect the situation.
  Hint: try $ pipenv lock --pre if it is a pre-release dependency.
Could not find a version that matches selenium==2.35.0,>=3.14.0
Tried: 0.9.2, 1.0.1, 1.0.3, 2.0.0, 2.0.1, 2.1.0, 2.2.0, 2.3.0, 2.4.0, 2.5.0, 2.6.0, 2.7.0, 2.8.0, 2.8.1, 2.9.0, 2.10.0, 2.11.0, 2.11.1, 2.12.0, 2.12.1, 2.13.0, 2.13.1, 2.14.0, 2.15.0, 2.16.0, 2.17.0, 2.18.0, 2.18.1, 2.19.0, 2.19.1, 2.20.0, 2.21.0, 2.21.1, 2.21.2, 2.21.3, 2.22.0, 2.22.1, 2.23.0, 2.24.0, 2.25.0, 2.26.0, 2.27.0, 2.28.0, 2.29.0, 2.30.0, 2.31.0, 2.32.0, 2.33.0, 2.34.0, 2.35.0, 2.36.0, 2.37.0, 2.37.1, 2.37.2, 2.38.0, 2.38.1, 2.38.2, 2.38.3, 2.38.4, 2.39.0, 2.40.0, 2.41.0, 2.42.0, 2.42.1, 2.43.0, 2.44.0, 2.45.0, 2.46.0, 2.46.1, 2.47.0, 2.47.1, 2.47.1, 2.47.2, 2.47.3, 2.47.3, 2.48.0, 2.48.0, 2.49.0, 2.49.0, 2.49.1, 2.49.1, 2.49.2, 2.49.2, 2.50.0, 2.50.0, 2.50.1, 2.50.1, 2.51.0, 2.51.1, 2.52.0, 2.53.0, 2.53.0, 2.53.1, 2.53.1, 2.53.2, 2.53.2, 2.53.3, 2.53.3, 2.53.4, 2.53.4, 2.53.5, 2.53.5, 2.53.6, 2.53.6, 3.0.0, 3.0.0, 3.0.1, 3.0.1, 3.0.2, 3.0.2, 3.3.0, 3.3.0, 3.3.1, 3.3.1, 3.3.2, 3.3.2, 3.3.3, 3.3.3, 3.4.0, 3.4.0, 3.4.1, 3.4.1, 3.4.2, 3.4.2, 3.4.3, 3.4.3, 3.5.0, 3.5.0, 3.6.0, 3.6.0, 3.7.0, 3.7.0, 3.8.0, 3.8.0, 3.8.1, 3.8.1, 3.9.0, 3.9.0, 3.10.0, 3.10.0, 3.11.0, 3.11.0, 3.12.0, 3.12.0, 3.13.0, 3.13.0, 3.14.0, 3.14.0, 3.14.1, 3.14.1
Skipped pre-versions: 2.0.dev0, 2.0.dev1, 2.0.dev2, 2.0.dev3, 2.0.dev4, 2.0.dev5, 2.0.dev6, 2.0.dev9138, 2.0.dev9212, 2.0.dev9231, 2.0.dev9284, 2.0.dev9306, 2.0.dev9307, 2.0.dev9310, 2.0.dev9338, 2.0.dev9340, 2.0.dev9341, 2.0.dev9429, 2.0a5, 2.0b2, 2.0b3.dev0, 2.0b3, 2.0b4.dev0, 2.0rc1, 2.0rc2, 2.0rc3, 3.0.0b1, 3.0.0b1, 3.0.0b2, 3.0.0b2, 3.0.0b3, 3.0.0b3
There are incompatible versions in the resolved dependencies.

/usr/local/lib/python2.7/site-packages/pipenv/vendor/vistir/compat.py:109: ResourceWarning: Implicitly cleaning up <TemporaryDirectory '/var/folders/94/1kxvhgmd2z52q1g150zmzzr14h6bqf/T/pipenv-muPkID-requirements'>
  warnings.warn(warn_message, ResourceWarning)

$ pipenv graph 
Huxley==0.5
  - jsonpickle [required: ==0.4.0, installed: 0.4.0]
  - Pillow [required: ==2.2.1, installed: 2.2.1]
  - plac [required: ==0.9.1, installed: 0.9.1]
  - selenium [required: ==2.35.0, installed: 2.35.0]
splinter==0.9.0
  - selenium [required: >=3.14.0, installed: 2.35.0]
urllib3==1.23

````
-------------------------------------------------------------------------------

<details><summary>$ pipenv --support</summary>

Pipenv version: `'2018.10.13'`

Pipenv location: `'/usr/local/lib/python2.7/site-packages/pipenv'`

Python location: `'/usr/local/opt/python/bin/python2.7'`

Python installations found:

  - `2.7.13`: `/usr/local/Cellar/python/2.7.13_1/Frameworks/Python.framework/Versions/2.7/bin/python2.7`
  - `2.7.10`: `/System/Library/Frameworks/Python.framework/Versions/2.7/bin/python2.7`
  - `2.6.9`: `/System/Library/Frameworks/Python.framework/Versions/2.6/bin/python2.6`
  - `2.7.10`: `/usr/bin/pythonw`
  - `2.7.10`: `/usr/bin/python`

PEP 508 Information:

```
{'implementation_name': 'cpython',
 'implementation_version': '0',
 'os_name': 'posix',
 'platform_machine': 'x86_64',
 'platform_python_implementation': 'CPython',
 'platform_release': '16.7.0',
 'platform_system': 'Darwin',
 'platform_version': 'Darwin Kernel Version 16.7.0: Thu Jun 21 20:07:39 PDT 2018; root:xnu-3789.73.14~1/RELEASE_X86_64',
 'python_full_version': '2.7.13',
 'python_version': '2.7',
 'sys_platform': 'darwin'}
```

System environment variables:

  - `TERM_PROGRAM_VERSION`
  - `LOGNAME`
  - `USER`
  - `PATH`
  - `HOME`
  - `PS1`
  - `TERM_PROGRAM`
  - `LANG`
  - `TERM`
  - `Apple_PubSub_Socket_Render`
  - `SHLVL`
  - `SECURITYSESSIONID`
  - `XPC_FLAGS`
  - `TMPDIR`
  - `TERM_SESSION_ID`
  - `XPC_SERVICE_NAME`
  - `SSH_AUTH_SOCK`
  - `SHELL`
  - `PYTHONDONTWRITEBYTECODE`
  - `PIP_PYTHON_PATH`
  - `_`
  - `OLDPWD`
  - `__CF_USER_TEXT_ENCODING`
  - `PWD`
  - `PIP_SHIMS_BASE_MODULE`

Pipenv–specific environment variables:

Debug–specific environment variables:

  - `PATH`: `/usr/local/opt/node@8/bin:~/.toolbox/bin/:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin:/Users/kkhanz/bin:/Applications/MySQLWorkbench.app/Contents/MacOS`
  - `SHELL`: `/bin/bash`
  - `LANG`: `en_US.UTF-8`
  - `PWD`: `/Users/kkhanz/Documents/programming/piptry`

---------------------------

Contents of `Pipfile` ('/Users/kkhanz/Documents/programming/piptry/Pipfile'):

```toml
[[source]]
url = "https://pypi.org/simple"
verify_ssl = true
name = "pypi"

[dev-packages]

[packages]
splinter = "*"
huxley = "*"

[requires]
python_version = "2.7"

```

Contents of `Pipfile.lock` ('/Users/kkhanz/Documents/programming/piptry/Pipfile.lock'):

```json
{
    "_meta": {
        "hash": {
            "sha256": "52f555bccab2f7552f73d41abb8b3f58ed353487dfd6240805f0dc422bb4283c"
        },
        "pipfile-spec": 6,
        "requires": {
            "python_version": "2.7"
        },
        "sources": [
            {
                "name": "pypi",
                "url": "https://pypi.org/simple",
                "verify_ssl": true
            }
        ]
    },
    "default": {
        "selenium": {
            "hashes": [
                "sha256:ab192cd046164c40fabcf44b47c66c8b12495142f4a69dcc55ea6eeef096e614",
                "sha256:fdb6b1143d8899e8a32e358ad05bf5d89a480dbac359dbbd341592aa8696dcd1"
            ],
            "version": "==3.14.1"
        },
        "splinter": {
            "hashes": [
                "sha256:08c6ed2209ce0520c2f898ad30822ba8f52ed9226bda10c0253c5ea9fe418146",
                "sha256:c8442149e03b0e4c3cf721c4dc1fb00feb1faf52872f85921e07b0cbde467e93"
            ],
            "index": "pypi",
            "version": "==0.9.0"
        },
        "urllib3": {
            "hashes": [
                "sha256:a68ac5e15e76e7e5dd2b8f94007233e01effe3e50e8daddf69acfd81cb686baf",
                "sha256:b5725a0bd4ba422ab0e66e89e030c806576753ea3ee08554382c14e685d117b5"
            ],
            "version": "==1.23"
        }
    },
    "develop": {}
}

```
</details>

**Comment by uranusjr** [comment]
We are already aware of this phenomenon, however the way Python packaging works induces difficulties to introducing a solution. The best we can do now is to show the warning message you see.

**Comment by matteius** [comment]
I am closing this report because I cannot reproduce it on `pipenv==2022.1.8` which is now partly because of the example bug report, has a python2.7 version requirement--I tried to run with a python3 version but down stream of `huxley` which was last published in 2014, requires `plac==0.9.1` which seems to not be py3 compatible.   I suggest that @ik-nitk if you find this is still an issue, check the recent versions of pipenv and see if this is still an issue, if so check recent issue reports to see if there is already a report of it.  Thanks!
